resources/js/src/app/components/itemList/filter/FindologicItemFilter.spec.ts

```diff
@@ -0,0 +1,351 @@
+import Vuex from 'vuex';
+import { shallowMount, mount, createLocalVue } from '@vue/test-utils';
+import { Facet, ColorFacet, State } from '../../../shared/interfaces';
+import { Store } from 'vuex';
+import FindologicItemFilter from './FindologicItemFilter.vue';
+import VueCompositionAPI from '@vue/composition-api';
+import UrlBuilder from '../../../shared/UrlBuilder';
+
+const localVue = createLocalVue();
+
+localVue.use(Vuex);
+localVue.use(VueCompositionAPI);
+
+window.ceresTranslate = key => key;
+
+describe('FindologicItemFilter', () => {
+
+  let store: Store<State>;
+
+  beforeEach(() => {
+    store = new Vuex.Store({});
+  });
+
+  it('Shows the facet name', () => {
+    const facet: Facet = {
+      cssClass: '',
+      findologicFilterType: 'select',
+      id: 'test',
+      isMain: false,
+      itemCount: 3,
+      name: 'Facet name',
+      noAvailableFiltersText: '',
+      select: 'multiple',
+      type: '',
+      values: [
+        {
+          count: 9,
+          id: '20',
+          image: '',
+          items: [],
+          name: '22220',
+          position: 'item',
+          selected: false
+        },
+        {
+          count: 1,
+          id: '21',
+          image: '',
+          items: [],
+          name: '22221',
+          position: 'item',
+          selected: false
+        }
+      ]
+    };
+
+    store.state.itemList = {
+      isLoading: false,
+      selectedFacets: [],
+      facets: [facet]
+    };
+
+    const wrapper = shallowMount(FindologicItemFilter, { propsData: { 'facet': facet, 'filtersPerRow': 3 }, store, localVue });
+
+    expect(wrapper.element.querySelector(':scope > div.h3')!.innerHTML).toBe('Facet name');
+  });
+
+  it('Shows the noAvailableFiltersText when facet has no values', () => {
+    const dropdownFacet: Facet = {
```

Comment:
Each of these facets should be in a separate test. Similar to PHP you can have something like a data provider and test for each case. Here is an example of how this may look like:

```ts
function facetProvider() {
  return [
    {
      facet: {
        cssClass: '',
        findologicFilterType: 'select',
        id: 'test',
        isMain: false,
        itemCount: 0,
        name: 'Facet name',
        noAvailableFiltersText: 'No values available for this dropdown',
        select: 'multiple',
        type: '',
        values: []
      } as Facet,
      expectedNoAvailableFiltersTextMessage: 'No values available for this dropdown'
    }
  ];
}

it.each(facetProvider())('testname', (data) => {
  // ...test...
  console.log(data.facet);
  console.log(data.expectedNoAvailableFiltersTextMessage);
});
```

Reply:
Please do this for all tests where this would apply.